packages/classic/src/payments.ts

```diff
@@ -0,0 +1,51 @@
+import { addPaymentHandler } from '@devvit/payments';
+import { type RedisClient } from '@devvit/public-api';
+import { DateTime } from 'luxon';
+
+class PaymentsRepo {
+  #redis: RedisClient;
+  constructor(redis: RedisClient) {
+    this.#redis = redis;
+  }
+  async addHardcoreModeLifetimeAccess(userId: string) {
+    await this.#redis.set(`hardcore-mode-lifetime-access:${userId}`, `1`);
+  }
+  async addHardcoreMode7DayAccess(userId: string) {
+    const sevenDaysFromNow = DateTime.now().plus({ days: 7 });
+    await this.#redis.set(`hardcore-mode-seven-day-access:${userId}`, sevenDaysFromNow.toISO());
```

Comment:
The 7 days should be on top of any existing active access. If I have 2 more days before my access ends and I purchase another 7 days, I should have 9 total days left.

This implementation would result in me losing out on 2 access days.